tests/testthat/setup-options.R

```diff
@@ -0,0 +1,17 @@
+opts_partial_match_old <- list(
+  warnPartialMatchDollar = getOption("warnPartialMatchDollar"),
+  warnPartialMatchArgs = getOption("warnPartialMatchArgs"),
+  warnPartialMatchAttr = getOption("warnPartialMatchAttr")
+)
+opts_partial_match_new <- list(
+  warnPartialMatchDollar = TRUE,
```

Comment:
Should we also set `warn = 1`?

It might help to debug when more than 10 are presented and it was on Hadley's original comment on {rcmdcheck} and {devtools}

`?options`
> If warn is zero (the default) warnings are stored until the top–level function returns. If 10 or fewer warnings were signalled they will be printed otherwise a message saying how many were signalled...
> If warn is one, warnings are printed as they occur. 

```suggestion
  warn = 1,
  warnPartialMatchDollar = TRUE,
```

Reply:
Can you please elaborate more what's the difference?

I was experimenting with `warn = 2` and abandoned this idea to keep this as warning. This has also undesirable and hard to control side-effects.

Currently we should have the following (temporarily re-introduced the old error):
```
$> devtools::test()
ℹ Testing teal.slice
✔ | F W  S  OK | Context
✔ |          9 | calls_combine_by                                                                                                                                                  
(...)                                                                                                                                  
✔ |   1     16 | FilteredDataset                                                                                                                                                   
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Warning (test-FilteredDataset.R:138:3): ui_add is pure virtual
partial argument match of 'rege' to 'regexp'
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
✔ |         39 | FilterState                                                                                                                                                       
(...)             
```

With `warn = 1` it's the same. So I don't see any difference - at least from that perspective.

Reply:
You're right, {testthat} catches all of them, I probably had a bug `1:10` as I was seeing fewer warnings.

```r
testthat::test_that("warnings", {
  (\() {
    lapply(1:100, \(.x) warning(glue::glue("warning {.x}")))
    invisible()
  })()
})
```